File: pdu/python/hako_srv_msgs/pdu_cdr_conv_RegisterClientRequest.py
from .pdu_pytype_RegisterClientRequest import RegisterClientRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_RegisterClientRequest(writer: CdrWriter, src: RegisterClientRequest):
    writer.write_uint32(src.dummy)


def cdr_body_to_py_RegisterClientRequest(reader: CdrReader, dst: RegisterClientRequest):
    dst.dummy = reader.read_uint32()
    return dst


def py_to_cdr_RegisterClientRequest(src: RegisterClientRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_RegisterClientRequest(writer, src)
    return writer.bytes()


def cdr_to_py_RegisterClientRequest(cdr_payload) -> RegisterClientRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = RegisterClientRequest()
    return cdr_body_to_py_RegisterClientRequest(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_RegisterClientRequest.js
import { RegisterClientRequest } from './pdu_jstype_RegisterClientRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduRegisterClientRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { RegisterClientRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint32(src.dummy);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { RegisterClientRequest } dst
     * @returns { RegisterClientRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.dummy = reader.read_uint32();
        return dst;
    }

    /**
     * @param { RegisterClientRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { RegisterClientRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new RegisterClientRequest());
    }
}
